src/lib.rs

```diff
@@ -123,3 +123,46 @@ pub mod groups;
 pub mod blocks;
 pub mod borrowed_iter;
 pub mod reader;
+
+use std::collections::{HashSet, HashMap};
+use std::io::{Seek, Read};
+
+pub fn get_objs_and_deps<R, F>(reader: &mut OsmPbfReader<R>, mut pred: F) -> Result<HashMap<OsmId, OsmObj>>
+    where R: Read + Seek,
+          F: FnMut(&OsmObj) -> bool
+{
+    let mut finished = false;
+    let mut dependencies = HashSet::new();
+    let mut objects = HashMap::new();
+    while !finished{
+        finished = true;
+        for block in reader.primitive_blocks(){
+            let block = try!(block);
+            for obj in blocks::iter(&block) {
+                if dependencies.contains(&obj.id()) || pred(&obj){
+                    match obj{
+                        OsmObj::Relation(ref rel) => {
+                            for reference in &rel.refs{
+                                if dependencies.insert(reference.member) {
+                                    finished = false;
+                                }
+                            }
```

Comment:
For fun, you can

```rust
if rel.refs.iter().fold(false, |accu, r| deps.insert(*r) || accu) {
    finished = false;
}
```

You can't use `Iterator::any()` as it is lazy.

If you like, you can use it, but if you prefere the iterative version, that's OK for me.